css/cssom-view/scroll-behavior-005.html

```diff
@@ -0,0 +1,141 @@
+<!DOCTYPE html>
+<title>Testing smooth scrolling on an element</title>
+<meta name="timeout" content="long"/>
+<link rel="author" title="Frédéric Wang" href="mailto:fwang@igalia.com">
+<link rel="help" href="https://drafts.csswg.org/cssom-view/#propdef-scroll-behavior">
+<link rel="help" href="https://drafts.csswg.org/cssom-view/#scrolling-box">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="support/scroll-behavior.js"></script>
+<style>
+  .scrollable {
+    overflow: auto;
+    width: 400px;
+    height: 200px;
+    scroll-behavior: smooth;
+  }
+</style>
+<div id="log">
+</div>
+<div id="overflowNode" class="scrollable">
+  <div style="width: 2000px; height: 1000px; background: linear-gradient(135deg, red, green);">
+    <span style="display: inline-block; width: 500px; height: 250px;"></span><span id="elementToReveal" style="display: inline-block; vertical-align: -15px; width: 10px; height: 15px; background: black;"></span>
+  </div>
+</div>
+<script>
+  ["scroll", "scrollTo", "scrollBy", "scrollIntoView"].forEach(function(scrollFunction) {
+    promise_test(() => {
+      return new Promise(function(resolve) {
+        resetScroll(overflowNode);
+        var finalLeft = 500;
+        var finalTop = 250;
+        // Note: exact smooth scrolling behavior is not specified by CSSOM View. This test relies on the minimal assumption that the element scrolls less than 1/4 of the total distance between two frames.
+        var maxDeltaLeft = finalLeft / 4;
+        var maxDeltaTop = finalTop / 4;
```

Comment:
I wonder if these constraints will cause flakiness when run as layout tests, particularly on debug builds where you might occasionally have a long gap between frames. You might need to relax these to 1/2 the distance or even 3/4 of the distance.

Reply:
Yes, I was wondering that too. I guess I can change to 3/4 or even just remove that constraint. BTW the current uploaded patch for WebKit fails the test but passes for 1/2, I have a local version to make the scroll smoother.